src/common/src/config.rs

```diff
@@ -539,12 +539,14 @@ mod default {
             std::thread::available_parallelism().unwrap().get()
         }
 
+        #[cfg(madsim)]
         pub fn total_memory_available_bytes() -> usize {
-            use sysinfo::{System, SystemExt};
-
-            let mut sys = System::new();
-            sys.refresh_memory();
-            sys.total_memory() as usize
+            16 * 1024 * 1024 * 1024
+        }
+        #[cfg(not(madsim))]
```

Comment:
nit: I would prefer an additional blank line.

```suggestion

        #[cfg(not(madsim))]
```